### PR TITLE
feat: Add last join exit to outputs csv so you can understand why earned fees may be 0

### PR DESCRIPTION
**Pull request by Tritium-VLK**
Waiting release of bal_addresses 0.9.1 and update from the temp version tag used here in the requirements before this should be merged.

https://github.com/BalancerMaxis/bal_addresses/pull/233

Example Run: https://github.com/BalancerMaxis/protocol_fee_allocator/actions/runs/8332269118
Example Ouput: #148

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -5,4 +5,4 @@ pycoingecko==3.1.0
 pandas==2.0.3
 simplejson==3.19.2
 
-git+https://github.com/BalancerMaxis/bal_addresses@0.8.4
```
Comment (Tritium-VLK) [inline requirements.txt:8]: This needs to be updated to 0.9.1 once released, should remain in draft until then.

**Comment by Tritium-VLK** [comment]
See: https://github.com/BalancerMaxis/protocol_fee_allocator/actions/runs/8378227265 for a final test of the whole shabang with the new v 0.9.1 being used for bal_addresses.

**Comment by Tritium-VLK** [comment]
Results from above run: https://github.com/BalancerMaxis/protocol_fee_allocator/pull/151

File: fee_allocator/accounting/distribution.py [hunk 2]
```diff
@@ -171,6 +174,28 @@ def re_distribute_incentives(
...
+    for pool_id, incentive_data in incentives.items():
+        results[pool_id] = incentive_data
+        timestamp = q.get_last_join_exit(pool_id)
+        try:
+            timestamp = q.get_last_join_exit(pool_id)
+        except:
+            results[pool_id]["last_join_exit"] = "Error fetching"
+            continue
+        gmt_time = datetime.datetime.utcfromtimestamp(timestamp)
+        human_time = gmt_time.strftime('%Y-%m-%d %H:%M:%S GMT')
```
Comment (Xeonus) [inline fee_allocator/accounting/distribution.py:193]: GMT might be a bit annoying for time sorting. I suggest to use something else like YYYY-MM-DDTHH:MM:SSZ so the output is similar to 2023-03-23T18:10:55+00:00 as an example

**Review by Xeonus — CHANGES_REQUESTED** [review]
Think of a better output format for the date string so it is easily usable with other tools (and sorting)

File: fee_allocator/accounting/distribution.py (hunk 2 above)
Comment (Tritium-VLK) [reply fee_allocator/accounting/distribution.py:193]: Sounds good.  Do you want the specific/exact time string you have specified to make life easier, or can I just change  GMT here to %Z?

Comment (Xeonus) [reply fee_allocator/accounting/distribution.py:193]: The latter sounds good

**Comment by Tritium-VLK** [comment]
See https://github.com/BalancerMaxis/bal_addresses/issues/239 for some suggestions for bal_addresses that came from trying to make this work.  Not directly related to join/exit timing, but related to when core pools are added to config vs when gauges are added.

**Comment by Tritium-VLK** [comment]
Found some other problems and fixed them %Z and %z are still not returning proper timezone formats.  I probably have to rethink how time is being handled here. I don't think any of this is timezone aware and everythign is just always done in GMT so asking for a timezone is hard because it never existed.

@Xeonus what is your real requirement here.  Do you want the timezone to change or just for GMT to be expressed as +00:00.  Can add that as a fixed string.   Will take another look at this monday once we have had more time to think about it.

**Comment by Xeonus** [comment]
Exactly, +00:00 would suffice and is easy to parse by js mappers.

**Comment by Tritium-VLK** [comment]
ok @Xeonus how does this look?

**Comment by Xeonus** [comment]
Thanks ser, looking good

**Review by Xeonus — APPROVED** [review]

**Pull request merged**